app/components/new-file/new-file.component.ts

```diff
@@ -21,58 +22,46 @@ export class NewFileResult {
 })
 export class NewFileComponent implements ModalComponent, AfterViewInit {
     set modalInfo(modalInfo: ModalInfo) {
-        let plugins: string[][] = modalInfo.data;
-        let availablePlugins: any = { subLists: {} };
+        const plugins: PluginData[] = modalInfo.data;
+        // This code is extremely evil but necessary.
```

Comment:
That is a fair point that I agree with. However, I don't think that I have time now to fix it. This would be a good issue to tackle when I refactor core to do descriptions according to npm. What I think would be ideal is to have a proper serialization mechanism in core for plugins.